test/integration/e2eIntegrationTests.coffee

```diff
@@ -1369,3 +1369,34 @@ describe "e2e Integration Tests", ->
                 done err
               else
                 done()
+
+
+
+  describe "Certificate Watcher", ->
+
+    afterEach (done) ->
+      server.stop ->
+        done()
+
+    it "Should Listen for certificate changes and restart the server with new certificate", (done) ->
+
+      server.start httpPort: 5001, httpsPort: 5000, ->
+        testUtils.setupTestKeystore (keystore) ->
+
+          # change the certificate file
+          fs.writeFile 'test/resources/certificate-watcher/test.crt', 'New certificate content. should trigger certificate reload', (err) ->
+            if err
+              return console.log(err)
+          
+          setTimeout (->
+            # reset the certificate
+            fs.createReadStream('test/resources/certificate-watcher/testBackup.crt').pipe(fs.createWriteStream('test/resources/certificate-watcher/test.crt'));
+
+            Keystore.findOne {}, (err, keystore) ->
+              done(err) if err
+              keystore.cert.data.should.be.exactly 'New certificate content. should trigger certificate reload'
+              done()
+            return
+          ), 1500
+        
+        
```

Comment:
As we chatted about, remove the test until we can figure out a good way to test this... :/